src/main/webapp/WEB-INF/templates/system/common/navigation.ftl

```diff
@@ -0,0 +1,166 @@
+<#include "/templates/system/common/ice.ftl"/>
+
+<#macro navigation
+<#---->url="/site/website"
+<#---->navElementClass=""
+<#---->showNavElement=true
+<#---->containerElement="ul"
+<#---->containerElementClass=""
+<#---->itemWrapperElement="li"
+<#---->itemWrapperClass=""
+<#---->itemWrapperActiveClass="active"
+<#---->itemWrapperAttributes={}
+<#---->itemClass=""
+<#---->itemActiveClass="active"
+<#---->itemAttributes={}
+<#---->hasSubItemItemClass=""
+<#---->hasSubItemWrapperClass=""
+<#---->hasSubItemItemAttributes={}
+<#---->subItemClass=""
+<#---->subItemClassPrefix="nav-level"
+<#---->subItemAttributes={}
+<#---->subItemWrapperClass=""
+<#---->subItemWrapperClassPrefix=""
+<#---->subItemContainerClass=""
+<#---->depth=1
+<#---->includeRoot=true
+<#---->inlineRootWithImmediateChildren=true
+>
+  <#assign navTree = navTreeBuilder.getNavTree(url, depth, Request.pageUrl)/>
+  <#-- navElement will be rendered if: showNavElement = true -->
+  <#if showNavElement != false><nav <#if navElementClass != ''>class="${navElementClass}"</#if>></#if>
+
+  <#-- containerElement will be rendered if containerElement has no empty value, and includeRoot is set to true -->
+  <#-- if includeRoot is false, then the first container will be the one containing the root subitems -->
+  <#if (containerElement != "") && (includeRoot)><${containerElement} <#if containerElementClass != ''>class="${containerElementClass}"</#if>></#if>
+    <@navigationItem
```

Comment:
does this mean we loose support for custom macros? https://docs.craftercms.org/en/4.0/developers/projects/engine/api/templating-api.html#using-custom-macros

Reply:
I don't think so, but I don't fully understand how the custom macros were meant to work (do they override ours so that even ours internally call the site overrides?). 

We're not doing anything to impede you from creating your own macros. These new ones are much less broken apart, though — being only two macros instead of 5.

We've made this new one very configurable testing it against 5+ popular web frameworks being able to accommodate their specifics successfully.